react-effects/my-app/src/List.js

```diff
@@ -0,0 +1,44 @@
+/* eslint-disable no-unused-vars -- REMOVE ME */
+import { useEffect, useState } from 'react';
+import readItems from './read';
+
+export default function List() {
+  const [isLoading, setIsLoading] = useState(true);
+  const [items, setItems] = useState([]);
+  const [error, setError] = useState();
+
+  // Your code here:
+  //  - When the component mounts:
+  //    - Read the items using `readItems` and update state so the list displays
+  //    - Handle errors from `readItems`
+
+  useEffect(() => {
+    async function renderitems() {
+      try {
+        setItems(await readItems());
+        setIsLoading(false);
+      } catch (err) {
+        setError(err);
```

Comment:
If an error happens, isLoading will still be true so you'll display the loading message instead of the error message. You need to set isLoading to false after an error, too.

The easiest way to do this is to just call `setIsLoading(false);` inside the `finally` clause of the `try/catch` block.